nexus/src/app/sagas/switch_port_settings_clear.rs

```diff
@@ -181,3 +204,185 @@ async fn spa_undo_clear_switch_port_settings(
+}
+
+async fn spa_clear_switch_port_bgp_settings(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+    let nexus = osagactx.nexus();
+    let opctx = crate::context::op_context_for_saga_action(
+        &sagactx,
+        &params.serialized_authn,
+    );
+
+    let orig_port_settings_id = sagactx
+        .lookup::<Option<Uuid>>("original_switch_port_settings_id")
+        .map_err(|e| {
+            ActionError::action_failed(format!(
+                "original port settings id lookup: {e}"
+            ))
+        })?;
+
+    let id = match orig_port_settings_id {
+        Some(id) => id,
+        None => return Ok(()),
+    };
+
+    let settings = nexus
+        .switch_port_settings_get(&opctx, &NameOrId::Id(id))
+        .await
+        .map_err(ActionError::action_failed)?;
+
+    let mg_client: Arc<mg_admin_client::Client> =
+        select_mg_client(&sagactx).await.map_err(|e| {
+            ActionError::action_failed(format!("select mg client (undo): {e}"))
+        })?;
+
+    for peer in settings.bgp_peers {
+        let config = nexus
+            .bgp_config_get(&opctx, peer.bgp_config_id.into())
+            .await
+            .map_err(|e| {
+                ActionError::action_failed(format!("delete bgp config: {e}"))
+            })?;
+
+        mg_client
+            .inner
+            .delete_neighbor(&DeleteNeighborRequest {
+                asn: *config.asn,
+                addr: peer.addr.ip(),
+            })
+            .await
+            .map_err(|e| {
+                ActionError::action_failed(format!("delete neighbor: {e}"))
+            })?;
+    }
+
+    Ok(())
+}
+
+async fn spa_undo_clear_switch_port_bgp_settings(
+    sagactx: NexusActionContext,
+) -> Result<(), Error> {
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+    let nexus = osagactx.nexus();
+    let opctx = crate::context::op_context_for_saga_action(
+        &sagactx,
+        &params.serialized_authn,
+    );
+
+    let orig_port_settings_id =
+        sagactx.lookup::<Option<Uuid>>("original_switch_port_settings_id")?;
+
+    let id = match orig_port_settings_id {
+        Some(id) => id,
+        None => return Ok(()),
+    };
+
+    let settings =
+        nexus.switch_port_settings_get(&opctx, &NameOrId::Id(id)).await?;
+
+    Ok(ensure_switch_port_bgp_settings(sagactx, settings).await?)
+}
+
+async fn spa_clear_switch_port_bootstore_network_settings(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    let nexus = sagactx.user_data().nexus();
+    let params = sagactx.saga_params::<Params>()?;
+    let opctx = crate::context::op_context_for_saga_action(
+        &sagactx,
+        &params.serialized_authn,
+    );
+
+    // Just choosing the sled agent associated with switch0 for no reason.
+    let sa = switch_sled_agent(SwitchLocation::Switch0, &sagactx).await?;
```

Comment:
This doesn't necessarily need fixing in this PR. However, once we get to doing all this with an RPW we'll probably want to be more dynamic about picking sled-agents in case one is offline.